HPVDeepSeek/HPVDeepSeekGenotyping.wdl

```diff
@@ -0,0 +1,1281 @@
+        File per_base_coverage = "~{prefix}.per_base_coverage.txt"
+    }
+
+    runtime {
+        cpu: cpu
+        memory: "~{memory_gb} GiB"
+        disks: "local-disk ~{disk_size_gb} HDD"
+        docker: "us-central1-docker.pkg.dev/broad-gp-hydrogen/hydrogen-dockers/kockan/hds@sha256:56f964695f08ddb74e3a29c63c3bc902334c1ddd735735cc98ba6d6a4212285c"
+    }
+}
+
+workflow HPVDeepSeekGenotyping {
+    input {
+        String output_basename
+        File? ubam
+        File? r1_fastq
+        File? r2_fastq
+        File human_snp_targets_bed
+        File reference
+        File reference_fai
+        File reference_dict
+        File bwa_idx_amb
+        File bwa_idx_ann
+        File bwa_idx_bwt
+        File bwa_idx_pac
+        File bwa_idx_sa
+        File capture_targets_bed
+        File bait_interval_list
+        File target_interval_list
+        String bait_set_name
+        String read_group_id
+        String read_group_sample_name
+        String read_group_library_name = "LB_TEST"
+        String read_group_platform = "ILLUMINA"
+        String read_group_platform_unit = "PU_TEST"
+        String read_group_description = "KAPA_TE"
+    }
+
+    call VerifyPipelineInputs {
+        input:
+            bam = ubam,
+            r1_fastq = r1_fastq,
+            r2_fastq = r2_fastq
+    }
+
+    if(VerifyPipelineInputs.fastq_run) {
+        call FastQC as PreTrimmedFastQC {
+            input:
+                r1_fastq = select_first([r1_fastq]),
+                r2_fastq = select_first([r2_fastq])
+        }
+
+        call FastqToUbam {
+            input:
+                r1_fastq = select_first([r1_fastq]),
+                r2_fastq = select_first([r2_fastq]),
+                output_basename = output_basename,
+                read_group_id = read_group_id,
+                read_group_sample_name = read_group_sample_name,
+                read_group_library_name = read_group_library_name,
+                read_group_platform = read_group_platform,
+                read_group_platform_unit = read_group_platform_unit,
+                read_group_description = read_group_description
+        }
+    }
+
+    File ubam_to_use = select_first([ubam, FastqToUbam.ubam])
+
+    call ExtractUMIs {
+        input:
+            input_ubam = ubam_to_use,
+            output_basename = output_basename
+    }
+
+    call UmiExtractedBamToFastq {
+        input:
+            umi_extracted_bam = ExtractUMIs.umi_extracted_bam,
+            output_basename = output_basename
+    }
+
+    call TrimAndFilter {
+        input:
+            fastq1 = UmiExtractedBamToFastq.umi_extracted_fastq1,
+            fastq2 = UmiExtractedBamToFastq.umi_extracted_fastq2,
+            output_basename = output_basename
+    }
+
+    call FastQC as PostTrimmedFastQC {
+        input:
+            r1_fastq = TrimAndFilter.fastq1_trimmed,
+            r2_fastq = TrimAndFilter.fastq2_trimmed
+    }
+
+    call BwaMem as AlignReads {
```

Comment:
is it really necessary to perform this pre-consensus alignment and collect all of these pre-consensus metrics?  or could you go straight to calling molecular consensus and skip all of this?